src/output_module.h

```diff
@@ -46,6 +46,8 @@ struct output_module {
 	int (*set_volume)(float);
 	int (*get_mute)(int *);
 	int (*set_mute)(int);
+
+	struct output_module *next;
```

Comment:
Given that this struct is part of the public interface how to implement an output module, the linked list pointer is probably not part of what we'd like to expose.

So the maintenance of the list should be outside of this struct, even if it is more cumbersome than an intrusive data structure. Public interfaces should be as lean and clean as possible and not leak internal implementation details.